src/StringTools/OpportunisticIntern.cs

```diff
@@ -0,0 +1,295 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.StringTools
+{
+    /// <summary>
+    /// Interns strings by holding weak references.
+    /// </summary>
+    internal sealed class OpportunisticIntern : IDisposable
```

Comment:
Should this class be static?

Reply:
This code is gone now. I believe it was implemented like this (state in instance fields + a static `Instance` getter) for testability, i.e. all state can be easily reset by re-creating the singleton instance.